service/routers/decks.js

```diff
@@ -0,0 +1,184 @@
+import { Router } from 'express'
+import { body } from 'express-validator'
+import { User } from '../models/User.js'
+
+const decksRouter = Router()
+
+const getDecks = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    console.log(`Other data from the token ${role}`)
+    try {
+      const user = await User.findById(req.params.id)
+      if (user) {
+        res.send(user.decks)
+      } else {
+        res.sendStatus(404)
+      }
+    } catch (err) {
+      console.log(`${getDecks.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else {
+    res.status(404).send('you are not authorized to access this route')
+  }
+  
+}
+
+const createDeck = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    const newDeck = req.body
+    try {
+      const user = await User.findById(req.params.id)
```

Comment:
Pay attention to the id you're using here

Reply:
I have taken userId of params to insert a new deck inside the user decks. Also, I have corrected my spelling mistake and changed req.params.id to req.params.userId